waspc/data/Generator/templates/server/src/auth/providers/oauth/user.ts

```diff
@@ -0,0 +1,90 @@
+{{={= =}=}}
+import { HttpError } from 'wasp/server'
+import {
+  type ProviderId,
+  createUser,
+  sanitizeAndSerializeProviderData,
+  validateAndGetUserFields,
+  createProviderId,
+} from 'wasp/auth/utils'
+import { type {= authEntityUpper =} } from 'wasp/entities'
+import { prisma } from 'wasp/server'
+import { type UserSignupFields, type ProviderConfig } from 'wasp/auth/providers/types'
+import { getRedirectUriForOneTimeCode, getRedirectUriForError } from './redirect'
+import { tokenStore } from './oneTimeCode'
+
+export async function finishOAuthFlowAndGetRedirectUri(
+  provider: ProviderConfig,
+  providerProfile: unknown,
+  providerUserId: string,
+  userSignupFields: UserSignupFields | undefined,
+): Promise<URL> {
+  const providerId = createProviderId(provider.id, providerUserId);
+
+  const authId = await getAuthIdFromProviderDetails(providerId, providerProfile, userSignupFields);
+
+  if (!authId) {
```

Comment:
Explicit check if you want

Reply:
This function always returns a value or it throws an error, so I've removed this check.